### PR TITLE
story 1251 - Copy and edit a project

### PR DESCRIPTION
**Pull request by GSuma21**

### CODE REVIEW

File: projects/lib-project/src/lib/dock/components/certificates/certificates.component.html [hunk 1]
```diff
@@ -275,11 +272,13 @@
...
-                                    <input matSliderThumb [value]="item?.values ? item?.values :item?.evidence_details.min_no_of_evidences"  [id]="item.id"  matInput #taskRange (change)="changeEvidenceCriteriaValue(taskRange.value,1,item)" [style.pointer-events]="viewOnly ? 'none' : 'auto'">
+                                <mat-slider class="red-slider" [min]="1" [id]="item.id" [disabled]="libProjectService.projectData.formMeta.taskEvidenceSelected[item.id] == 0 || !libProjectService.projectData.formMeta.taskEvidenceSelected[item.id]" [max]="maximunNumberOfEvedence" step="1" showTickMarks discrete style="width: 100%;" [style.pointer-events]="viewOnly ? 'none' : 'auto'" (slideend)="onSliderInput($event, item, item?.evidence_details.min_no_of_evidences, taskRange.value)"  (input)="onSliderInput($event, item, item?.evidence_details.min_no_of_evidences, taskRange.value)" (pointerup)="onSliderInput($event, item, item?.evidence_details.min_no_of_evidences, taskRange.value)">
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/dock/components/certificates/certificates.component.html]: what is this class for ?

Comment (ramkumar-ns-dev) [reply projects/lib-project/src/lib/dock/components/certificates/certificates.component.html]: class="red-slider"

File: projects/lib-project/src/lib/dock/view-module/layout/layout.component.ts [hunk 2]
```diff
@@ -186,6 +186,18 @@ export class LayoutComponent {
         )
         break;
       }
+      case "COPY_AND_EDIT":{
+        this.libProjectService.copyAndCreateProject().subscribe((res:any)=>{
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/dock/view-module/layout/layout.component.ts]: why subscription is not unsubscribed ?

File: projects/lib-project/src/lib/dock/components/certificates/certificates.component.ts [hunk 3]
```diff
@@ -664,8 +666,12 @@ export class CertificatesComponent implements OnInit, OnDestroy,AfterViewInit{
     // this.libProjectService.projectData.certificate.criteria.conditions.C3.conditions[item.id].value = taskCriteria > 0 ? criterialValue : 0;
   }
 
-  changeEvidenceCriteriaValue(criterialValue:any,taskCriteria:any,item:any) {
-    this.libProjectService.projectData.certificate.criteria.conditions.C3.conditions[item.id].value = taskCriteria > 0 ? criterialValue : 0;
+  changeEvidenceCriteriaValue(criterialValue:any,taskCriteria:any,item:any,minTaskEvidence:any) {
```
Comment (ramkumar-ns-dev) [inline projects/lib-project/src/lib/dock/components/certificates/certificates.component.ts:670]: what is this function will do ?

**Review by ramkumar-pacewisdom — CHANGES_REQUESTED** [review]

File: projects/lib-project/src/lib/dock/components/certificates/certificates.component.html (hunk 1 above)
Comment (GSuma21) [reply projects/lib-project/src/lib/dock/components/certificates/certificates.component.html]: removed unwanted class

File: projects/lib-project/src/lib/dock/view-module/layout/layout.component.ts (hunk 2 above)
Comment (GSuma21) [reply projects/lib-project/src/lib/dock/view-module/layout/layout.component.ts]: added subscription

File: projects/lib-project/src/lib/dock/components/certificates/certificates.component.ts (hunk 3 above)
Comment (GSuma21) [reply projects/lib-project/src/lib/dock/components/certificates/certificates.component.ts:670]: this function add the value for  this.libProjectService.projectData.certificate.criteria.conditions.C3.conditions[item.id].value
according to the condition 
if criterialValue is less than min number of evidence then value should add as minEvidence made this condition change

**Review by ramkumar-pacewisdom — APPROVED** [review]

**Pull request merged**